src/hotspot/share/services/mallocTracker.cpp

```diff
@@ -70,6 +70,24 @@ size_t MallocMemorySnapshot::total_arena() const {
   return amount;
 }
 
+size_t MallocMemorySnapshot::thread_count() const {
+  if (ThreadStackTracker::track_as_vm()) {
+    return _thread_count;
+  } else {
+    assert(_thread_count == 0, "_thread_count can not be used if ThreadStackTracker::track_as_vm() == false");
+    MallocMemorySnapshot* s = const_cast<MallocMemorySnapshot*>(this);
```

Comment:
Why is this necessary instead of calling the method directly?

Reply:
Fixed, thanks